frame/include/bli_extern_defs.h

```diff
@@ -35,16 +35,26 @@
 #ifndef BLIS_EXTERN_DEFS_H
 #define BLIS_EXTERN_DEFS_H
 
-extern obj_t BLIS_TWO;
-extern obj_t BLIS_ONE;
-//extern obj_t BLIS_ONE_HALF;
-extern obj_t BLIS_ZERO;
-//extern obj_t BLIS_MINUS_ONE_HALF;
-extern obj_t BLIS_MINUS_ONE;
-extern obj_t BLIS_MINUS_TWO;
-
-extern thrcomm_t BLIS_SINGLE_COMM;
-extern thrinfo_t BLIS_PACKM_SINGLE_THREADED;
-extern thrinfo_t BLIS_GEMM_SINGLE_THREADED;
+#if !defined(BLIS_ENABLE_SHARED) || !defined(_MSC_VER)
+#define BLIS_EXPORT
+#else
+#ifdef BLIS_BUILD_DLL
```

Comment:
@fgvanzee, I want this to be defined only at all build time and not when tests are built

Reply:
@isuruf It sounds like I need to provide a cpp macro for general use that is only defined when BLIS is being compiled, and is not enabled within `blis.h` for applications. Then you can branch off of that macro.

Reply:
@isuruf I've pushed commit 4b5437ec to `master`, which defines `BLIS_IS_BUILDING_LIBRARY` when BLIS is being built and leaves it undefined for application use cases (testsuite, example code, etc.)

Reply:
@fgvanzee, I tried master and the tests are using the macro as well.

Here's what I get on Linux with `make check V=1`
```
gcc -O3 -Wall -Wno-unused-function -Wfatal-errors -fPIC -std=c99 -D_POSIX_C_SOURCE=200112L -Iinclude/haswell -I./frame/3/ -I./frame/ind/ukernels/ -I./frame/1m/ -I./frame/1f/ -I./frame/1/ -I./frame/include -DBLIS_VERSION_STRING=\"0.4.1-11\" -DBLIS_IS_BUILDING_LIBRARY -c testsuite/src/test_axpyf.c -o obj/haswell/testsuite/test_axpyf.o
```

Reply:
Thanks for pointing this out. I previously fixed the individual test drivers' Makefiles, but forgot to fix the top-level `Makefile` when building BLAS test drivers and the BLIS testsuite. This should be fixed now in 1330d5c. You should no longer need to set `BUILD_FLAGS` to the empty string on line 27 of `.appveyor.yml`.